virttest/utils_backup.py

```diff
@@ -0,0 +1,297 @@
+            return True
+    utils_misc.wait_for(file_exists, 5)
+    return cp_xml
+
+
+def create_backup_xml(backup_params, disk_xml_list=None):
+    """
+    Create a backup xml
+
+    :param backup_params: Params of backup
+    :param disk_xml_list: A list of disk params used in backup xml
+    :return: The backup xml
+    """
+    backup_xml = BackupXML()
+    backup_mode = backup_params.get("backup_mode")
+    backup_incremental = backup_params.get("backup_incremental")
+    backup_server = backup_params.get("backup_server")
+    if backup_mode:
+        backup_xml.mode = backup_mode
+    if backup_incremental:
+        backup_xml.incremental = backup_incremental
+    if backup_server:
+        if type(backup_server) is not dict:
+            logging.error("Backup server tag should be defined by a dict.")
+            return
+        backup_xml.server = backup_server
+    backup_xml.set_disks(disk_xml_list)
+    backup_xml.xmltreefile.write()
+
+    def file_exists():
+        if os.path.exists(backup_xml.xml):
+            return True
+    utils_misc.wait_for(file_exists, 5)
+    return backup_xml
+
+
+def create_backup_disk_xml(backup_disk_params):
+    """
+    Create a disk xml which is a subelement of a backup xml
+
+    :param backup_disk_params: Params of disk xml
+    :return: The disk xml
+    """
+    backup_disk_xml = BackupXML.DiskXML()
+    disk_name = backup_disk_params.get("disk_name")
+    disk_type = backup_disk_params.get("disk_type")
+    enable_backup = backup_disk_params.get("enable_backup")
+    exportname = backup_disk_params.get("exportname")
+    exportbitmap = backup_disk_params.get("exportbitmap")
+    backup_target = backup_disk_params.get("backup_target")  # dict
+    backup_driver = backup_disk_params.get("backup_driver")  # dict
+    backup_scratch = backup_disk_params.get("backup_scratch")  # dict
+    if not disk_name:
+        logging.error("Disk name must be provided for backup disk xml.")
+        return
+    backup_disk_xml.name = disk_name
+    if disk_type:
+        backup_disk_xml.type = disk_type
+    if enable_backup:
+        backup_disk_xml.backup = enable_backup
+    if exportname:
+        backup_disk_xml.exportname = exportname
+    if exportbitmap:
+        backup_disk_xml.exportbitmap = exportbitmap
+    if backup_target:
+        if type(backup_target) is not dict:
+            logging.error("disk target tag should be defined by a dict.")
+            return
+        backup_disk_xml.target = backup_target
+    if backup_driver:
+        if type(backup_driver) is not dict:
+            logging.error("disk driver tag should be defined by a dict.")
+            return
+        backup_disk_xml.driver = backup_driver
+    if backup_scratch:
+        if type(backup_scratch) is not dict:
+            logging.error("disk scratch tag should be defined by a dict.")
+        backup_disk_xml.scratch = backup_scratch
+    backup_disk_xml.xmltreefile.write()
+
+    def file_exists():
```

Comment:
there are 3 places to define file_exists(), maybe pull out as common function

Reply:
directly use "utils_misc.wait_for(lambda: os.path.exists(cp_xml.xml), 5)" now, thx